[WIP] React native upgrade from version 0.63.3 to 0.71.4 ?

siddarthkay opened the pull request:
fixes #14386 
status: ready

siddarthkay committed:
react-native v0.63.3 -> v0.68.5 [Android]

siddarthkay committed:
react-native v0.63.3 -> v0.68.5 [IOS]

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :x: | c2b1ed18 | [#2](https://ci.status.im/job/status-mobile/job/prs/job/android-e2e/job/PR-15203/2/) | 2023-02-27 16:14:37 | ~1 min | `android-e2e` | [:page_facing_up:`log`](https://ci.status.im/job/status-mobile/job/prs/job/android-e2e/job/PR-15203/2/consoleText) |

siddarthkay committed:
ios - specific fixes

fix Podfile for `react-native-reanimated`

update `AppDelegate.mm` reference in xcodeProject

some more logs to `app-launched-listener`

upgrade RNN & add more logs

upgrade lib `react-native-reanimated`

to get rid of iOS Warning: RCTBridge required dispatch_sync to load REAModule

get rid of deprecated `bootstrapWithDelegate`

update yarn.lock :)

siddarthkay commented:
known side effect on IOS Side : https://github.com/status-im/status-mobile/issues/15206

jakubgs committed:
android - specific fixes

bump `react-native-hole-view`

use status-im org fork of `react-native-hole-view`

use status-im org fork of `react-native-mail`

use correct fork of `react-native-mail`

nix: use new flags for gradle call

Signed-off-by: Jakub Sokołowski <jakub@status.im>

nix: fix aapt2 derivationt trying to patch a text file

Signed-off-by: Jakub Sokołowski <jakub@status.im>

nix: make AAPT2 installPhase optional for macos

Signed-off-by: Jakub Sokołowski <jakub@status.im>

nix: fix gradle dependencies

Signed-off-by: Jakub Sokołowski <jakub@status.im>

nix: add missing gradle deps (NOT A REAL FIX)

Signed-off-by: Jakub Sokołowski <jakub@status.im>

nix: drop setting JAVA_HOME, breaks apkanalyzer

Signed-off-by: Jakub Sokołowski <jakub@status.im>

jakubgs commented:
Looks like the JAR for AAPT2 now also includes a `NOTICE` which has exec permissions:
```
+++ find /build/tmp.OnogTjWbML/ -type f -executable
++ for exe in `find $tmpDir/ -type f -executable`
++ type /build/tmp.OnogTjWbML/NOTICE
/build/tmp.OnogTjWbML/NOTICE is /build/tmp.OnogTjWbML/NOTICE
++ file /build/tmp.OnogTjWbML/NOTICE
/build/tmp.OnogTjWbML/NOTICE: ASCII text
+++ cat /nix/store/j061mvdxw70xz20r2r0lr5lk0rcsjf9i-gcc-wrapper-11.3.0/nix-support/dynamic-linker
++ patchelf --set-interpreter /nix/store/lkxc9m0lm2c9lxzhfxs4hsll6nx3fjsl-glibc-2.35-163/lib/ld-linux-x86-64.so.2 /build/tmp.OnogTjWbML/NOTICE
patchelf: not an ELF executable
```
Which would cause the derivation to try to `patchelf` a text file instead of an ELF binary:
https://github.com/status-im/status-mobile/blob/e01618595e2ad590b85a0171a714128382832c7d/nix/pkgs/aapt2/default.nix#L73-L76
Easy fix.

jakubgs commented:
Seems to me like this error:
```
Plugin [id: 'org.jetbrains.kotlin.jvm', version: '1.6.10'] was not found in any of the following sources:

- Gradle Core Plugins (plugin is not in 'org.gradle' namespace)
- Included Builds (None of the included builds contain this plugin)
- Plugin Repositories (could not resolve plugin artifact 'org.jetbrains.kotlin.jvm:org.jetbrains.kotlin.jvm.gradle.plugin:1.6.10')
  Searched in the following repositories:
    MavenRepo
    Google
    Gradle Central Plugin Repositor
```
Means that Gradle is not looking for the plugin files in Nix store. Which can mean at least two of those things:

1. Gradle 6 had all the necessary plugins already provided via `maven.repo.local` setting.
2. Gradle 6 was not using any plugins in the first place.

Which would mean that either:

1. Gradle 7 is not searching for plugins in Nix store path provided via `maven.repo.local`.
2. Gradle 7 now requires plugins which were previously not necessary.

Which means either we have to point it to the right path, or provide plugins in some alternative way.

jakubgs commented:
According to the docs:

>A plugin is simply any class that implements the [Plugin](https://docs.gradle.org/current/javadoc/org/gradle/api/Plugin.html) interface. Gradle provides the core plugins (e.g. JavaPlugin) as part of its distribution which means they are automatically resolved. However, non-core binary plugins need to be resolved before they can be applied. This can be achieved in a number of ways:
>
>* Including the plugin from the plugin portal or a [custom repository](https://docs.gradle.org/current/userguide/plugins.html#sec:custom_plugin_repositories) using the plugins DSL ([Applying plugins using DSL](https://docs.gradle.org/current/userguide/plugins.html#sec:plugins_block)).
>* Including the plugin from an external jar defined as a buildscript dependency (see [Applying plugins using the buildscript](https://docs.gradle.org/current/userguide/plugins.html#sec:applying_plugins_buildscript)).
>* Defining the plugin as a source file under the buildSrc directory in the project (see [Using buildSrc to extract func. logic](https://docs.gradle.org/current/userguide/organizing_gradle_projects.html#sec:build_sources)).
>* Defining the plugin as an inline class declaration inside a build script.

https://docs.gradle.org/current/userguide/plugins.html#sec:binary_plugins

jakubgs commented:
The way we do it, and it is normally done, is using the `buildscript` block:

>Binary plugins that have been published as external jar files can be added to a project by adding the plugin to the build script classpath and then applying the plugin. External jars can be added to the build script classpath using the `buildscript {}` block as described in [External dependencies for the build script](https://docs.gradle.org/current/userguide/tutorial_using_tasks.html#sec:build_script_external_dependencies).

https://docs.gradle.org/current/userguide/plugins.html#sec:applying_plugins_buildscript

jakubgs commented:
If I `strace` gradle I can see it loads some JARs for Kotlin plugin from Gradle nix folder itself:
```
 > grep kotlin result/strace.log | grep plugin | grep '.jar"' | sort -u
104   newfstatat(AT_FDCWD, "/nix/store/8fwhgsbmcbp064ki23l65shyvc9digwp-gradle-7.5/lib/gradle/lib/plugins/gradle-kotlin-dsl-tooling-builders-7.5.jar", {st_mode=S_IFREG|0444, st_size=126701, ...}, 0) = 0
31    newfstatat(AT_FDCWD, "/nix/store/8fwhgsbmcbp064ki23l65shyvc9digwp-gradle-7.5/lib/gradle/lib/kotlin-sam-with-receiver-compiler-plugin-1.6.21.jar", {st_mode=S_IFREG|0444, st_size=11710, ...}, 0) = 0
31    newfstatat(AT_FDCWD, "/nix/store/8fwhgsbmcbp064ki23l65shyvc9digwp-gradle-7.5/lib/gradle/lib/plugins/gradle-kotlin-dsl-provider-plugins-7.5.jar", {st_mode=S_IFREG|0444, st_size=186678, ...}, 0) = 0
31    newfstatat(AT_FDCWD, "/nix/store/8fwhgsbmcbp064ki23l65shyvc9digwp-gradle-7.5/lib/gradle/lib/plugins/gradle-kotlin-dsl-tooling-builders-7.5.jar", {st_mode=S_IFREG|0444, st_size=126701, ...}, 0) = 0
31    openat(AT_FDCWD, "/nix/store/8fwhgsbmcbp064ki23l65shyvc9digwp-gradle-7.5/lib/gradle/lib/kotlin-sam-with-receiver-compiler-plugin-1.6.21.jar", O_RDONLY) = 118
67    newfstatat(AT_FDCWD, "/nix/store/8fwhgsbmcbp064ki23l65shyvc9digwp-gradle-7.5/lib/gradle/lib/kotlin-sam-with-receiver-compiler-plugin-1.6.21.jar", {st_mode=S_IFREG|0444, st_size=11710, ...}, 0) = 0
67    newfstatat(AT_FDCWD, "/nix/store/8fwhgsbmcbp064ki23l65shyvc9digwp-gradle-7.5/lib/gradle/lib/plugins/gradle-kotlin-dsl-provider-plugins-7.5.jar", {st_mode=S_IFREG|0444, st_size=186678, ...}, 0) = 0
67    newfstatat(AT_FDCWD, "/nix/store/8fwhgsbmcbp064ki23l65shyvc9digwp-gradle-7.5/lib/gradle/lib/plugins/gradle-kotlin-dsl-tooling-builders-7.5.jar", {st_mode=S_IFREG|0444, st_size=126701, ...}, 0) = 0
67    openat(AT_FDCWD, "/nix/store/8fwhgsbmcbp064ki23l65shyvc9digwp-gradle-7.5/lib/gradle/lib/kotlin-sam-with-receiver-compiler-plugin-1.6.21.jar", O_RDONLY) = 118
67    openat(AT_FDCWD, "/nix/store/8fwhgsbmcbp064ki23l65shyvc9digwp-gradle-7.5/lib/gradle/lib/plugins/gradle-kotlin-dsl-provider-plugins-7.5.jar", O_RDONLY) = 135
67    openat(AT_FDCWD, "/nix/store/8fwhgsbmcbp064ki23l65shyvc9digwp-gradle-7.5/lib/gradle/lib/plugins/gradle-kotlin-dsl-provider-plugins-7.5.jar", O_RDONLY) = 157
67    openat(AT_FDCWD, "/nix/store/8fwhgsbmcbp064ki23l65shyvc9digwp-gradle-7.5/lib/gradle/lib/plugins/gradle-kotlin-dsl-tooling-builders-7.5.jar", O_RDONLY) = 135
67    openat(AT_FDCWD, "/nix/store/8fwhgsbmcbp064ki23l65shyvc9digwp-gradle-7.5/lib/gradle/lib/plugins/gradle-kotlin-dsl-tooling-builders-7.5.jar", O_RDONLY) = 180
```
But nothing else.

jakubgs commented:
I can't even find any attempt to read a file with the version we need in the filename:
```
 > grep kotlin result/strace.log | grep 1.6.10                         

```
Which is worrying.

jakubgs commented:
I found one possible issue in the error:
```
FAILURE: Build failed with an exception.

* Where:
Build file '/build/node_modules/react-native-gradle-plugin/build.gradle.kts' line: 11
```
They key thing here is that this file is `build.gradle.kts` and not `build.gradle`.
Right now when we patch Node dependencies we only take into account `build.gradle` files:
https://github.com/status-im/status-mobile/blob/064204256ea40d7a6d1c358ab06530752bbf6c00/nix/deps/nodejs-patched/default.nix#L28-L42

So that's one thing to fix, but it doesn't appear to make a difference.

jakubgs commented:
When I run Gradle with `--debug` flag I can see it's trying to download ``:
```
 > grep 'kotlin.*1.6.10' gradle.log
[DEBUG] [org.gradle.internal.operations.DefaultBuildOperationRunner] Build operation 'Download https://repo.maven.apache.org/maven2/org/jetbrains/kotlin/jvm/org.jetbrains.kotlin.jvm.gradle.plugin/1.6.10/org.jetbrains.kotlin.jvm.gradle.plugin-1.6.10.pom' started
[DEBUG] [org.gradle.internal.operations.DefaultBuildOperationRunner] Completing Build operation 'Download https://repo.maven.apache.org/maven2/org/jetbrains/kotlin/jvm/org.jetbrains.kotlin.jvm.gradle.plugin/1.6.10/org.jetbrains.kotlin.jvm.gradle.plugin-1.6.10.pom'
[DEBUG] [org.gradle.internal.operations.DefaultBuildOperationRunner] Build operation 'Download https://dl.google.com/dl/android/maven2/org/jetbrains/kotlin/jvm/org.jetbrains.kotlin.jvm.gradle.plugin/1.6.10/org.jetbrains.kotlin.jvm.gradle.plugin-1.6.10.pom' started
[DEBUG] [org.gradle.internal.operations.DefaultBuildOperationRunner] Completing Build operation 'Download https://dl.google.com/dl/android/maven2/org/jetbrains/kotlin/jvm/org.jetbrains.kotlin.jvm.gradle.plugin/1.6.10/org.jetbrains.kotlin.jvm.gradle.plugin-1.6.10.pom'
[DEBUG] [org.gradle.internal.operations.DefaultBuildOperationRunner] Build operation 'Download https://plugins.gradle.org/m2/org/jetbrains/kotlin/jvm/org.jetbrains.kotlin.jvm.gradle.plugin/1.6.10/org.jetbrains.kotlin.jvm.gradle.plugin-1.6.10.pom' started
[DEBUG] [org.gradle.internal.operations.DefaultBuildOperationRunner] Completing Build operation 'Download https://plugins.gradle.org/m2/org/jetbrains/kotlin/jvm/org.jetbrains.kotlin.jvm.gradle.plugin/1.6.10/org.jetbrains.kotlin.jvm.gradle.plugin-1.6.10.pom'
[DEBUG] [org.gradle.api.internal.artifacts.ivyservice.ivyresolve.RepositoryChainComponentMetaDataResolver] Attempting to resolve component for org.jetbrains.kotlin.jvm:org.jetbrains.kotlin.jvm.gradle.plugin:1.6.10 using repositories [MavenRepo, Google, Gradle Central Plugin Repository]
[DEBUG] [org.gradle.api.internal.artifacts.repositories.resolver.DefaultExternalResourceArtifactResolver] Loading https://repo.maven.apache.org/maven2/org/jetbrains/kotlin/jvm/org.jetbrains.kotlin.jvm.gradle.plugin/1.6.10/org.jetbrains.kotlin.jvm.gradle.plugin-1.6.10.pom
[DEBUG] [org.gradle.internal.resource.transfer.DefaultCacheAwareExternalResourceAccessor] Constructing external resource: https://repo.maven.apache.org/maven2/org/jetbrains/kotlin/jvm/org.jetbrains.kotlin.jvm.gradle.plugin/1.6.10/org.jetbrains.kotlin.jvm.gradle.plugin-1.6.10.pom
[DEBUG] [org.gradle.internal.resource.transport.http.HttpResourceAccessor] Constructing external resource: https://repo.maven.apache.org/maven2/org/jetbrains/kotlin/jvm/org.jetbrains.kotlin.jvm.gradle.plugin/1.6.10/org.jetbrains.kotlin.jvm.gradle.plugin-1.6.10.pom
[DEBUG] [org.gradle.internal.resource.transport.http.HttpClientHelper] Performing HTTP GET: https://repo.maven.apache.org/maven2/org/jetbrains/kotlin/jvm/org.jetbrains.kotlin.jvm.gradle.plugin/1.6.10/org.jetbrains.kotlin.jvm.gradle.plugin-1.6.10.pom
[DEBUG] [org.gradle.internal.operations.DefaultBuildOperationRunner] Build operation 'Download https://repo.maven.apache.org/maven2/org/jetbrains/kotlin/jvm/org.jetbrains.kotlin.jvm.gradle.plugin/1.6.10/org.jetbrains.kotlin.jvm.gradle.plugin-1.6.10.pom' completed
[DEBUG] [org.gradle.api.internal.artifacts.repositories.resolver.DefaultExternalResourceArtifactResolver] Loading https://dl.google.com/dl/android/maven2/org/jetbrains/kotlin/jvm/org.jetbrains.kotlin.jvm.gradle.plugin/1.6.10/org.jetbrains.kotlin.jvm.gradle.plugin-1.6.10.pom
[DEBUG] [org.gradle.internal.resource.transfer.DefaultCacheAwareExternalResourceAccessor] Constructing external resource: https://dl.google.com/dl/android/maven2/org/jetbrains/kotlin/jvm/org.jetbrains.kotlin.jvm.gradle.plugin/1.6.10/org.jetbrains.kotlin.jvm.gradle.plugin-1.6.10.pom
[DEBUG] [org.gradle.internal.resource.transport.http.HttpResourceAccessor] Constructing external resource: https://dl.google.com/dl/android/maven2/org/jetbrains/kotlin/jvm/org.jetbrains.kotlin.jvm.gradle.plugin/1.6.10/org.jetbrains.kotlin.jvm.gradle.plugin-1.6.10.pom
[DEBUG] [org.gradle.internal.resource.transport.http.HttpClientHelper] Performing HTTP GET: https://dl.google.com/dl/android/maven2/org/jetbrains/kotlin/jvm/org.jetbrains.kotlin.jvm.gradle.plugin/1.6.10/org.jetbrains.kotlin.jvm.gradle.plugin-1.6.10.pom
[DEBUG] [org.gradle.internal.operations.DefaultBuildOperationRunner] Build operation 'Download https://dl.google.com/dl/android/maven2/org/jetbrains/kotlin/jvm/org.jetbrains.kotlin.jvm.gradle.plugin/1.6.10/org.jetbrains.kotlin.jvm.gradle.plugin-1.6.10.pom' completed
[DEBUG] [org.gradle.api.internal.artifacts.repositories.resolver.DefaultExternalResourceArtifactResolver] Loading https://plugins.gradle.org/m2/org/jetbrains/kotlin/jvm/org.jetbrains.kotlin.jvm.gradle.plugin/1.6.10/org.jetbrains.kotlin.jvm.gradle.plugin-1.6.10.pom
[DEBUG] [org.gradle.internal.resource.transfer.DefaultCacheAwareExternalResourceAccessor] Constructing external resource: https://plugins.gradle.org/m2/org/jetbrains/kotlin/jvm/org.jetbrains.kotlin.jvm.gradle.plugin/1.6.10/org.jetbrains.kotlin.jvm.gradle.plugin-1.6.10.pom
[DEBUG] [org.gradle.internal.resource.transport.http.HttpResourceAccessor] Constructing external resource: https://plugins.gradle.org/m2/org/jetbrains/kotlin/jvm/org.jetbrains.kotlin.jvm.gradle.plugin/1.6.10/org.jetbrains.kotlin.jvm.gradle.plugin-1.6.10.pom
[DEBUG] [org.gradle.internal.resource.transport.http.HttpClientHelper] Performing HTTP GET: https://plugins.gradle.org/m2/org/jetbrains/kotlin/jvm/org.jetbrains.kotlin.jvm.gradle.plugin/1.6.10/org.jetbrains.kotlin.jvm.gradle.plugin-1.6.10.pom
[DEBUG] [org.gradle.internal.operations.DefaultBuildOperationRunner] Build operation 'Download https://plugins.gradle.org/m2/org/jetbrains/kotlin/jvm/org.jetbrains.kotlin.jvm.gradle.plugin/1.6.10/org.jetbrains.kotlin.jvm.gradle.plugin-1.6.10.pom' completed
[ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] Plugin [id: 'org.jetbrains.kotlin.jvm', version: '1.6.10'] was not found in any of the following sources:
[ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] - Plugin Repositories (could not resolve plugin artifact 'org.jetbrains.kotlin.jvm:org.jetbrains.kotlin.jvm.gradle.plugin:1.6.10')
```
Specifically, it tries the URL:
https://plugins.gradle.org/m2/org/jetbrains/kotlin/jvm/org.jetbrains.kotlin.jvm.gradle.plugin/1.6.10/org.jetbrains.kotlin.jvm.gradle.plugin-1.6.10.pom

Which does exist, but is just a POM without any JAR.

jakubgs commented:
Here we can see where Gradle is looking for the plugin POMs(and JARs):
```
3460518 newfstatat(AT_FDCWD, "/run/user/1000/tmp.F2sZZMLrXI/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-gradle-plugin-api", 0x7fdeab1f6c80, 0) = -1 ENOENT (No such file or directory)
3460518 newfstatat(AT_FDCWD, "/run/user/1000/tmp.F2sZZMLrXI/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-gradle-plugin-api", 0x7fdeab1f6e30, 0) = -1 ENOENT (No such file or directory)
3460518 newfstatat(AT_FDCWD, "/run/user/1000/tmp.F2sZZMLrXI/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-gradle-plugin-api/1.6.10", 0x7fdeab1f6c80, 0) = -1 ENOENT (No such file or directory)
3460518 newfstatat(AT_FDCWD, "/run/user/1000/tmp.F2sZZMLrXI/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-gradle-plugin-api/1.6.10/509dc9830a6c2f9cd395c0aff195f80309e76260", 0x7fdeab1f6c80, 0) = -1 ENOENT (No such file or directory)
3460518 newfstatat(AT_FDCWD, "/run/user/1000/tmp.F2sZZMLrXI/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-gradle-plugin-api/1.6.10/509dc9830a6c2f9cd395c0aff195f80309e76260/kotlin-gradle-plugin-api-1.6.10.pom", 0x7fdeab1f6d10, 0) = -1 ENOENT (No such file or directory)
3460518 newfstatat(AT_FDCWD, "/run/user/1000/tmp.F2sZZMLrXI/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-gradle-plugin-api/1.6.10/509dc9830a6c2f9cd395c0aff195f80309e76260/kotlin-gradle-plugin-api-1.6.10.pom", 0x7fdeab1f6e00, 0) = -1 ENOENT (No such file or directory)
```
The location for the POM is:
```
$GRADLE_USER_HOME/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-gradle-plugin-api/1.6.10/509dc9830a6c2f9cd395c0aff195f80309e76260/kotlin-gradle-plugin-api-1.6.10.pom
```
We can put it there, but what the hell is `509dc9830a6c2f9cd395c0aff195f80309e76260`.

jakubgs commented:
Apparently:

>You are comparing the SHA of file with SHA of url. The sha you are considering from gradle is the sha of url from where the jar is downloaded. gradle maintains sha of url for caching and avoid multiple downloads of jar in the dependency tree. As you know, sha will be unique so existing dependency jar will never download again until you change the dependency itself. Wheres the file sha is a kind of signature to ensure the content of file is not modified and it's as-is. Hope it's clear now.

https://stackoverflow.com/questions/58021095/whats-the-rule-of-the-sha1-code-directory-in-gradle-cache-path

So that's annoying, but doable.

jakubgs commented:
Aaaand there we have it:

>Gradle will use repositories at two different phases during your build.
>
>The first phase is when [configuring your build](https://docs.gradle.org/7.5/userguide/build_lifecycle.html#sec:build_phases) and loading the plugins it applied. To do that __Gradle will use a special set of repositories__.

And:

>__By default Gradle will use the [Gradle plugin portal](http://plugins.gradle.org/) to look for plugins.__
>
>However, for different reasons, there are plugins available in other, public or not, repositories. When a build requires one of these plugins, additional repositories need to be specified so that Gradle knows where to search.
>
>As the way to declare the repositories and what they are expected to contain depends on the way the plugin is applied, it is best to refer to [Custom Plugin Repositories](https://docs.gradle.org/7.5/userguide/plugins.html#sec:custom_plugin_repositories).

https://docs.gradle.org/7.5/userguide/declaring_repositories.html#sub:plugin-repos

So yes, Gradle does ignore our `mavenLocal` configuration and just fetches from http://plugins.gradle.org/ anyway.

Moronic.

jakubgs commented:
And indeed `react-native-gradle-plugin` does define its own `pluginManagement` section:
```
 > grep -A6 pluginManagement node_modules/react-native-gradle-plugin/settings.gradle.kts
pluginManagement {
    repositories {
        mavenCentral()
        google()
        gradlePluginPortal()
    }
}
```

jakubgs commented:
Interestingly, we already use this for `react-native-share`:
https://github.com/status-im/status-mobile/blob/1f1ab30a4ee2dbd9e3a5b9be1af2980479390d19/android/settings.gradle#L3-L8

And we use `mavenLocal`, although it's not documented.

jakubgs commented:
Another smaller issue is this missing dependency:
```
Could not determine the dependencies of task ':app:lintVitalReportRelease'.
> Could not resolve all task dependencies for configuration ':app:releaseRuntimeClasspath'.
   > Could not resolve com.facebook.conceal:conceal:1.1.3.
     Required by:
         project :app > project :react-native-keychain
      > No cached version of com.facebook.conceal:conceal:1.1.3 available for offline mode.
```
But that is due to how Gradle shows the dependency in the printed tree:
```
+--- project :react-native-keychain
|    +--- com.facebook.react:react-native:+ -> 0.68.5 (*)
|    \--- com.facebook.conceal:conceal:1.1.3 FAILED
```
It has some kind of issue resolving it, so it's listed as `FAILED`.

jakubgs commented:
Currently I can fix the build by manually running this:
```
nix/deps/gradle/add_package.sh org.jetbrains.kotlin.jvm:org.jetbrains.kotlin.jvm.gradle.plugin:1.6.10
```
But we need a proper way to include this dependency when running `make nix-update-gradle`.

jakubgs commented:
When I run:
```
./gradlew --no-daemon --console=plain react-native-gradle-plugin:buildEnvironment
```
I get:
```
Project 'react-native-gradle-plugin' not found in root project 'StatusIm'.
```
Which is interesting.

jakubgs commented:
What is this, do we actually use it, and why is it not in `packages.json` or `yarn.lock`?
```
 > ag -a react-native-orientation-locker 
src/react_native/orientation.cljs
2:  (:require ["react-native-orientation-locker" :default orientation :refer (useDeviceOrientationChange)]

src/mocks/js_dependencies.cljs
337:(def react-native-orientation-locker
406:    "react-native-orientation-locker"               react-native-orientation-locker
```

siddarthkay commented:
This seems to be introduced very recently by Omar and I have not rebased this PR so probably that's why our yarn.lock is outdated

jakubgs commented:
That's something new:
```
FAILURE: Build failed with an exception.

* Where:
Build file '/home/jakubgs/work/status-mobile/node_modules/react-native-mail/android/build.gradle' line: 4

* What went wrong:
A problem occurred evaluating project ':react-native-mail'.
> Unsupported value: 31. Format must be one of:
  - android-31
  - android-31-ext2
  - android-T
  - vendorName:addonName:31
```
I think I'm done for today.

siddarthkay commented:
> That's something new:
> 
> ```
> 
> FAILURE: Build failed with an exception.
> 
> 
> 
> * Where:
> 
> Build file '/home/jakubgs/work/status-mobile/node_modules/react-native-mail/android/build.gradle' line: 4
> 
> 
> 
> * What went wrong:
> 
> A problem occurred evaluating project ':react-native-mail'.
> 
> > Unsupported value: 31. Format must be one of:
> 
>   - android-31
> 
>   - android-31-ext2
> 
>   - android-T
> 
>   - vendorName:addonName:31
> 
> ```
> 
> I think I'm done for today.



I know this issue, it's fixed in my fork, unfortunately I'm out for dinner, I'll pull the latest code and try to recreate it locally

jakubgs commented:
I assume the 31 it's referring to is the platform version:
https://github.com/status-im/status-mobile/blob/1f1ab30a4ee2dbd9e3a5b9be1af2980479390d19/nix/pkgs/android-sdk/compose.nix#L14

jakubgs commented:
The key issue right now is that `make nix-update-gradle` does not find `org.jetbrains.kotlin.jvm:org.jetbrains.kotlin.jvm.gradle.plugin:1.6.10`:
```
FAILURE: Build failed with an exception.

* Where:
Build file '/build/node_modules/react-native-gradle-plugin/build.gradle.kts' line: 11

* What went wrong:
Plugin [id: 'org.jetbrains.kotlin.jvm', version: '1.6.10'] was not found in any of the following sources:

- Gradle Core Plugins (plugin is not in 'org.gradle' namespace)
- Included Builds (None of the included builds contain this plugin)
- Plugin Repositories (could not resolve plugin artifact 'org.jetbrains.kotlin.jvm:org.jetbrains.kotlin.jvm.gradle.plugin:1.6.10')
  Searched in the following repositories:
    MavenLocal(file:/nix/store/c92ljvh74r04f1b9cd3s8kw0p9jsy88w-status-mobile-maven-deps/)
    MavenLocal2(file:/nix/store/c92ljvh74r04f1b9cd3s8kw0p9jsy88w-status-mobile-maven-deps/)
    MavenLocal3(file:/nix/store/c92ljvh74r04f1b9cd3s8kw0p9jsy88w-status-mobile-maven-deps/)
```
But if we check for dependencies of `react-native-gradle-plugin` we get:
```
Project 'react-native-gradle-plugin' not found in root project 'StatusIm'.
```
Which means that Gradle is not detecting this as a sub-project of our main project.

jakubgs commented:
To figure out where this was coming from:
```
Execution failed for task ':react-native-dialogs:extractReleaseAnnotations'.
> Could not resolve all files for configuration ':react-native-dialogs:detachedConfiguration1'.
   > Could not resolve com.android.tools.lint:lint-gradle:30.3.0.
     Required by:
         project :react-native-dialogs
      > No cached version of com.android.tools.lint:lint-gradle:30.3.0 available for offline mode.
      > No cached version of com.android.tools.lint:lint-gradle:30.3.0 available for offline mode.
```
I ran Gradle with `--scan` and identified that the dependency on `com.android.tools.lint:lint-gradle:30.3.0` comes from something called `detachedConfiguration1`:

![image](https://user-images.githubusercontent.com/2212681/223551361-ed6e3985-9efd-4c0f-a2c0-c271edbd309f.png)

Which is probably why it doesn't show up in `:dependencies` and `:buildEnvironment` tasks.

siddarthkay committed:
CI - ios debugging

show pods and revert the Podfile change

show me what you got in Pods dir

log contents & permissions of IOS PR Pods on fail

patch glog in `patchReactNativePhase`

update gems

remove degenerate pods

add comment todo to remove patch fix later

maybe xcode needs to be cleaned up

get rid of `react-native-codegen`

add default flags to Podfile

:deterministic_uuids => false

ios: pods update, xcode config fix

:deterministic_uuids => false breaks stuff locally

try `make release-ios-pr` when ios build fails

on failure try harder

fix quotes

install pods via makefile

get rid of useless steps

print rn environment info on CI

check rn-info on failure as well

make release-ios-pr

bring back UseModernBuildSystem flag

try if `make release-ios-pr` works on CI macos

try building ios instead of archiving in Prebundle

target ios

fix make

try building scheme StatusIm as well

show CI xcode buildSettings

add `verbose` flag to fastlane ios builds

add verbose to xcargs

use nix shell for rn-info

use script block

xcode-clean on prep stage

can't xcode-select --reset without sudo

dont print info in pre build stage

try on a specific host

no label

update label

`__apply_Xcode_12_5_M1_post_install_workaround`

try and clean xcode better

get rid of entire DerivedData folder

get rid of header_search_path

xcode something something

upgrade nodejs to 18.x

status-im-auto commented:
#### :heavy_check_mark: [status-mobile/prs/android/PR-15203#40](https://ci.status.im/job/status-mobile/job/prs/job/android/job/PR-15203/40/) :small_blue_diamond: ~8 min 1 sec :small_blue_diamond: 9c20b7f2 :small_blue_diamond: [:package: android package](https://status-im-mobile-prs.ams3.digitaloceanspaces.com/StatusIm-Mobile-230309-125315-9c20b7-pr15203-universal.apk)

siddarthkay committed:
(0.71.4) RN IOS Upgrade

enable new architecture

don't patchBuildIdPhase & patchJavaPhase TEMP

build without fabric

siddarthkay commented:
The IOS build archive fails with error 
```
/Users/jenkins/workspace/status-mobile_prs_ios_PR-15203@2/ios/Pods/Target Support Files/Pods-Status-StatusImPR/Pods-Status-StatusImPR.release.xcconfig:1:1: error: unable to open configuration settings file
/Users/jenkins/workspace/status-mobile_prs_ios_PR-15203@2/ios/Pods/Target Support Files/Pods-Status-StatusImPR/Pods-Status-StatusImPR.release.xcconfig:1:1: error: unable to open configuration settings file
/Users/jenkins/workspace/status-mobile_prs_ios_PR-15203@2/ios/Pods/Target Support Files/Pods-Status-StatusImPR/Pods-Status-StatusImPR.release.xcconfig:1:1: error: unable to open configuration settings file
```
This means that `Pods-Status-StatusImPR.release.xcconfig` file either does not exist at `ios/Pods/Target Support Files/` directory or it has incorrect access permissions.

On further investigation I found that `Target Support Files` directory inside Pods is empty on the CI MacOS host

```
[2023-03-10T02:54:29.031Z] + ls -l 'ios/Pods/Target Support Files/'
[2023-03-10T02:54:29.033Z] total 0
```

I also noticed this logged message which should have thrown an error 
`sed: can't read s/\@ac_cv_have_libgflags\@/0/: No such file or directory`

On further investigating I found this sed command was inside `podspec.json` of `glog` Pod on my local environment.
And the `glog` directory was on CI 
```
[2023-03-10T02:54:28.485Z] + ls -l ios/Pods/
[2023-03-10T02:54:28.485Z] total 0
[2023-03-10T02:54:28.485Z] drwxr-xr-x   5 jenkins  jenkins   160 Mar 10 03:54 CryptoSwift
[2023-03-10T02:54:28.485Z] drwxr-xr-x   5 jenkins  jenkins   160 Mar 10 03:54 DoubleConversion
[2023-03-10T02:54:28.485Z] drwxr-xr-x   5 jenkins  jenkins   160 Mar 10 03:54 HMSegmentedControl
[2023-03-10T02:54:28.485Z] drwxr-xr-x   2 jenkins  jenkins    64 Mar 10 03:51 Headers
[2023-03-10T02:54:28.485Z] drwxr-xr-x   5 jenkins  jenkins   160 Mar 10 03:54 Keycard
[2023-03-10T02:54:28.485Z] drwxr-xr-x  75 jenkins  jenkins  2400 Mar 10 03:52 Local Podspecs
[2023-03-10T02:54:28.485Z] drwxr-xr-x   5 jenkins  jenkins   160 Mar 10 03:54 RCT-Folly
[2023-03-10T02:54:28.485Z] drwxr-xr-x   7 jenkins  jenkins   224 Mar 10 03:54 SDWebImage
[2023-03-10T02:54:28.485Z] drwxr-xr-x   5 jenkins  jenkins   160 Mar 10 03:54 SDWebImageWebPCoder
[2023-03-10T02:54:28.485Z] drwxr-xr-x   6 jenkins  jenkins   192 Mar 10 03:54 SQLCipher
[2023-03-10T02:54:28.485Z] drwxr-xr-x   5 jenkins  jenkins   160 Mar 10 03:54 SSZipArchive
[2023-03-10T02:54:28.485Z] drwxr-xr-x   5 jenkins  jenkins   160 Mar 10 03:54 TOCropViewController
[2023-03-10T02:54:28.485Z] drwxr-xr-x   2 jenkins  jenkins    64 Mar 10 03:51 Target Support Files
[2023-03-10T02:54:28.485Z] drwxr-xr-x   9 jenkins  jenkins   288 Mar 10 03:54 boost
[2023-03-10T02:54:28.485Z] drwxr-xr-x   6 jenkins  jenkins   192 Mar 10 03:54 fmt
[2023-03-10T02:54:28.485Z] drwxr-xr-x   5 jenkins  jenkins   160 Mar 10 03:54 secp256k1
```
so this means that pod install process was incomplete.

siddarthkay commented:
On further investigating I found this log which confirmed my suspicion regarding pods being incomplete

```
[2023-03-10T02:53:56.199Z] Installing glog (0.3.5)
[2023-03-10T02:53:56.199Z] 
[2023-03-10T02:53:56.199Z] [!] Error installing glog
[2023-03-10T02:53:56.199Z] [!] /nix/store/qixpggqvxi0f9ldnxpbfbb7qmnyzpjsa-git-2.37.3/bin/git clone https://github.com/google/glog.git /tmp/d20230310-62106-a8h3a5 --template= --single-branch --depth 1 --branch v0.3.5
[2023-03-10T02:53:56.199Z] 
[2023-03-10T02:53:56.199Z] Cloning into '/tmp/d20230310-62106-a8h3a5'...
[2023-03-10T02:53:56.199Z] fatal: unable to access 'https://github.com/google/glog.git/': error:16000069:STORE routines::unregistered scheme
```
This looks like a network or a firewall Issue but I am not sure maybe @jakubgs will know.

jakubgs commented:
Firewall issues normally result in either a timeout or a rejected connection.
The `error:16000069:STORE routines::unregistered scheme` error is a certificate validation error.
This might be related to age of CA certificates from `nixpkgs` we are using.

jakubgs commented:
This is not a full proper fix, but it should avoid weird issues with `pod install` in calls that use unnecessarily pure shells:

* https://github.com/status-im/status-mobile/pull/15301
* https://github.com/status-im/status-jenkins-lib/pull/59

jakubgs commented:
It appears there's an issue with Cocoapods `glog` package being patched by React-Native at installation time:
```
patching file config.sub
sed: can't read s/\@ac_cv_have_libgflags\@/0/: No such file or directory
```
Which was apparently fixed in RN `v0.70.0`:

* https://github.com/facebook/react-native/issues/33966
* https://github.com/facebook/react-native/commit/4a7e4b9ca6ef4fb52611b6c3cb788f624d1f81a4

But that doesn't really help us since we are on `0.68.5`.

siddarthkay commented:
![nothing-seems-to-work-weve-tried-everything-saturday-night-live](https://user-images.githubusercontent.com/64726664/224624265-3358ddf8-5987-4360-9238-f70f93cb8100.gif)

siddarthkay commented:
hmm Thanks @flexsurfer, this is worth looking into :)

siddarthkay commented:
I remember making no change to the targets specifically to get the build working locally.

I looked into both the targets and tried to spot any significant difference in their settings.
Apart from their names and icons I could only spot a few differences worth experimenting with and on updating them the jenkins build still fails and consistently gives out the same error message.

I also tried to perform an Xcode build by changing the target and it was also successful locally.
![Screenshot 2023-03-13 at 6 51 46 PM](https://user-images.githubusercontent.com/64726664/224714873-d318dc06-12a9-44d7-b71f-8c7d2c2f22e2.png)

So I'm guessing thats probably not the root cause of our failure here.

Looking more into the error message on this file : `RCTNativeAnimatedModule.mm` we have not even enabled new architecture yet so probably this file means nothing to us but still is causing us soo much pain :) 
Its probably related to this change right here in `ios/StatusIm/AppDelegate.mm` : https://github.com/status-im/status-mobile/pull/15203/files#diff-5c4e88a7dbd54ad6622c1e63b419d4a6fe1dcac878b723fd5dde1cffc2287c1fR38

siddarthkay commented:
To further find differences in environments I did `npx react-native info` on both my local and CI macos

This is relevant output from CI macos : 

```
System:
[2023-03-15T02:05:38.557Z]     OS: macOS 13.1
[2023-03-15T02:05:38.557Z]     CPU: (8) arm64 Apple M1
[2023-03-15T02:05:38.557Z]     Memory: 8.88 GB / 16.00 GB
[2023-03-15T02:05:38.557Z]     Shell: 5.8.1 - /bin/zsh
[2023-03-15T02:05:38.557Z]   Binaries:
[2023-03-15T02:05:38.557Z]     Node: 16.19.0 - /opt/homebrew/bin/node
[2023-03-15T02:05:38.557Z]     Yarn: 1.22.19 - /opt/homebrew/bin/yarn
[2023-03-15T02:05:38.557Z]     npm: 8.19.3 - /opt/homebrew/bin/npm
[2023-03-15T02:05:38.557Z]     Watchman: Not Found
[2023-03-15T02:05:38.557Z]   Managers:
[2023-03-15T02:05:38.557Z]     CocoaPods: 1.7.5 - /usr/local/bin/pod
[2023-03-15T02:05:38.557Z]   SDKs:
[2023-03-15T02:05:38.557Z]     iOS SDK:
[2023-03-15T02:05:38.557Z]       Platforms: DriverKit 22.2, iOS 16.2, macOS 13.1, tvOS 16.1, watchOS 9.1
[2023-03-15T02:05:38.557Z]     Android SDK: Not Found
[2023-03-15T02:05:38.557Z]   IDEs:
[2023-03-15T02:05:38.557Z]     Android Studio: Not Found
[2023-03-15T02:05:38.557Z]     Xcode: 14.2/14C18 - /usr/bin/xcodebuild
[2023-03-15T02:05:38.557Z]   Languages:
[2023-03-15T02:05:38.557Z]     Java: 19.0.1 - /usr/bin/javac
[2023-03-15T02:05:38.557Z]   npmPackages:
[2023-03-15T02:05:38.557Z]     @react-native-community/cli: Not Found
[2023-03-15T02:05:38.557Z]     react: Not Found
[2023-03-15T02:05:38.557Z]     react-native: Not Found
[2023-03-15T02:05:38.557Z]     react-native-macos: Not Found
[2023-03-15T02:05:38.557Z]   npmGlobalPackages:
[2023-03-15T02:05:38.557Z]     *react-native*: Not Found
```

and this is relevant output from my macos : 

```
System:
    OS: macOS 13.2.1
    CPU: (8) x64 Apple M2
    Memory: 789.17 MB / 24.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 19.4.0 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 9.2.0 - /usr/local/bin/npm
    Watchman: Not Found
  Managers:
    CocoaPods: 1.11.3 - /Users/siddarthkumar/.rbenv/shims/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 22.2, iOS 16.2, macOS 13.1, tvOS 16.1, watchOS 9.1
    Android SDK: Not Found
  IDEs:
    Android Studio: 2022.1 AI-221.6008.13.2211.9514443
    Xcode: 14.2/14C18 - /usr/bin/xcodebuild
  Languages:
    Java: 19.0.1 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.2 => 17.0.2 
    react-native: 0.68.5 => 0.68.5 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

siddarthkay commented:
The differences are :
- `macOS 13.1` on CI and `macOS 13.2.1` locally ( probably shouldn't matter )
- `NodeJs: 16.19.0` on CI and `NodeJs: 19.4.0` locally (does not matter because we do not use environment `nodejs` we get it via `nix`)
- `npm: 8.19.3`  on CI and `npm: 9.2.0` locally (does not matter because we do not use environment `npm` we get it via `nix`)
- `CocoaPods: 1.7.5` on CI and `CocoaPods: 1.11.3 locally` ->  _this is a bit suspicious_

flexsurfer commented:
just wondering if you did run `RCT_NEW_ARCH_ENABLED=1 pod install` locally

flexsurfer commented:
also wondering if you run locally with `make run-ios` or from xcode

jakubgs commented:
Our CI hosts were using an old 1.7.5 version of Cocoapods so I upgraded that manually:
```
 > a ci-slave-macos --become-user admin -o -a '/usr/local/bin/pod --version'
macm1-01.he-eu-fsn1.ci.devel | CHANGED | rc=0 | (stdout) 1.11.3
maci7-01.ms-eu-dublin.ci.devel | CHANGED | rc=0 | (stdout) 1.11.3
maci7-02.ms-eu-dublin.ci.devel | CHANGED | rc=0 | (stdout) 1.11.3
maci7-03.ms-eu-dublin.ci.devel | CHANGED | rc=0 | (stdout) 1.11.3
macm1-01.ih-eu-mda1.ci.devel | CHANGED | rc=0 | (stdout) 1.11.3
macm1-02.he-eu-fsn1.ci.devel | CHANGED | rc=0 | (stdout) 1.11.3
```
This way we can confirm if the issue is the pods version, and if it is we'll need a fix for that too.

siddarthkay commented:
@flexsurfer  no I did not do this : RCT_NEW_ARCH_ENABLED=1 pod install
and make run-ios and builing from xcode both work for me locally, 
We think its old cocoapods issue and that will resolve this

siddarthkay commented:
change log : https://github.com/facebook/react-native/blob/main/CHANGELOG.md#v0690
upgrade helper : https://react-native-community.github.io/upgrade-helper/?from=0.68.5&to=0.69.8

My Previous worries were 

- upgrading `reactjs` from `17.x` to `18.x` which requires us to upgrade `reagent library` version which only partially supports `react js18`
ref : https://www.metosin.fi/blog/reagent-towards-react-18/
someone tried upgrading reagent to use `reactjs 18` and faced these DX issues : 
https://github.com/reagent-project/reagent/issues/579

- possible compatibility issues with our navigation library `react-native-navigation`
ref : https://github.com/wix/react-native-navigation/issues/7466

- possible pain with `bundled hermes` now being the default 
ref : https://reactnative.dev/architecture/bundled-hermes

My previous approach was to get a stable build of 0.68.x merged into develop and then move forward with dealing with the pains of further upgrading. 

At this point we might as well give upgrading to `0.69.8` a shot.

flexsurfer commented:
> Question: why are we upgrading to an unsupported version?

good question i also think we should go straight to 0.71 and new architecture

siddarthkay committed:
(0.69.8) - RN IOS Upgrade

siddarthkay commented:
@jakubgs `react-native: 0.69.8` also has the same bug on Jenkins MacOS host 🥵
ref : https://ci.status.im/blue/organizations/jenkins/status-mobile%2Fprs%2Fios/detail/PR-15203/150/pipeline/#step-72-log-625

```log
[2023-03-17T06:23:00.873Z] info Fetching system and libraries information...
[2023-03-17T06:23:01.200Z] System:
[2023-03-17T06:23:01.200Z]     OS: macOS
[2023-03-17T06:23:01.200Z]     CPU: (8) arm64 Apple M1
[2023-03-17T06:23:01.200Z]     Memory: 4.76 GB / 16.00 GB
[2023-03-17T06:23:01.200Z]     Shell: 5.1.16 - /nix/store/d8hcl0rdr85qjhk5hwwpk5mmz3yf93cc-bash-interactive-5.1-p16/bin/bash
[2023-03-17T06:23:01.200Z]   Binaries:
[2023-03-17T06:23:01.200Z]     Node: 18.9.1 - /nix/store/pxwkyggw18a39jp40g82xcmrs7xb9dpv-nodejs-18.9.1/bin/node
[2023-03-17T06:23:01.200Z]     Yarn: 1.22.19 - /nix/store/cbpkq785in3gmxgkw5vqagz6jfvbg4gr-yarn-1.22.19/bin/yarn
[2023-03-17T06:23:01.200Z]     npm: 8.19.1 - /nix/store/pxwkyggw18a39jp40g82xcmrs7xb9dpv-nodejs-18.9.1/bin/npm
[2023-03-17T06:23:01.200Z]     Watchman: 4.9.0 - /nix/store/77m93viygxsjmzyy1zqcyglz29r3pd3y-watchman-4.9.0/bin/watchman
[2023-03-17T06:23:01.200Z]   Managers:
[2023-03-17T06:23:01.200Z]     CocoaPods: 1.12.0 - /nix/store/n5k6vp57np0pxxqhzcyc4wwddvasp1vp-cocoapods-1.12.0/bin/pod
[2023-03-17T06:23:01.200Z]   SDKs:
[2023-03-17T06:23:01.200Z]     iOS SDK:
[2023-03-17T06:23:01.200Z]       Platforms: DriverKit 22.2, iOS 16.2, macOS 13.1, tvOS 16.1, watchOS 9.1
[2023-03-17T06:23:01.200Z]     Android SDK: Not Found
[2023-03-17T06:23:01.200Z]   IDEs:
[2023-03-17T06:23:01.200Z]     Android Studio: Not Found
[2023-03-17T06:23:01.200Z]     Xcode: 14.2/14C18 - /nix/store/i4qw4s2n4650cwkzmfxq1zm3yav26x1r-xcode-wrapper-13.3-plus/bin/xcodebuild
[2023-03-17T06:23:01.200Z]   Languages:
[2023-03-17T06:23:01.200Z]     Java: Not Found
[2023-03-17T06:23:01.200Z]   npmPackages:
[2023-03-17T06:23:01.200Z]     @react-native-community/cli: Not Found
[2023-03-17T06:23:01.200Z]     react: 18.0.0 => 18.0.0 
[2023-03-17T06:23:01.200Z]     react-native: 0.69.8 => 0.69.8 
[2023-03-17T06:23:01.200Z]     react-native-macos: Not Found
[2023-03-17T06:23:01.200Z]   npmGlobalPackages:
[2023-03-17T06:23:01.200Z]     *react-native*: Not Found
```

siddarthkay commented:
@jakubgs `react-native: 0.69.8` also has the same bug on Jenkins MacOS host 🥵
ref : https://ci.status.im/blue/organizations/jenkins/status-mobile%2Fprs%2Fios/detail/PR-15203/150/pipeline/#step-72-log-625

```log
[2023-03-17T06:23:00.873Z] info Fetching system and libraries information...
[2023-03-17T06:23:01.200Z] System:
[2023-03-17T06:23:01.200Z]     OS: macOS
[2023-03-17T06:23:01.200Z]     CPU: (8) arm64 Apple M1
[2023-03-17T06:23:01.200Z]     Memory: 4.76 GB / 16.00 GB
[2023-03-17T06:23:01.200Z]     Shell: 5.1.16 - /nix/store/d8hcl0rdr85qjhk5hwwpk5mmz3yf93cc-bash-interactive-5.1-p16/bin/bash
[2023-03-17T06:23:01.200Z]   Binaries:
[2023-03-17T06:23:01.200Z]     Node: 18.9.1 - /nix/store/pxwkyggw18a39jp40g82xcmrs7xb9dpv-nodejs-18.9.1/bin/node
[2023-03-17T06:23:01.200Z]     Yarn: 1.22.19 - /nix/store/cbpkq785in3gmxgkw5vqagz6jfvbg4gr-yarn-1.22.19/bin/yarn
[2023-03-17T06:23:01.200Z]     npm: 8.19.1 - /nix/store/pxwkyggw18a39jp40g82xcmrs7xb9dpv-nodejs-18.9.1/bin/npm
[2023-03-17T06:23:01.200Z]     Watchman: 4.9.0 - /nix/store/77m93viygxsjmzyy1zqcyglz29r3pd3y-watchman-4.9.0/bin/watchman
[2023-03-17T06:23:01.200Z]   Managers:
[2023-03-17T06:23:01.200Z]     CocoaPods: 1.12.0 - /nix/store/n5k6vp57np0pxxqhzcyc4wwddvasp1vp-cocoapods-1.12.0/bin/pod
[2023-03-17T06:23:01.200Z]   SDKs:
[2023-03-17T06:23:01.200Z]     iOS SDK:
[2023-03-17T06:23:01.200Z]       Platforms: DriverKit 22.2, iOS 16.2, macOS 13.1, tvOS 16.1, watchOS 9.1
[2023-03-17T06:23:01.200Z]     Android SDK: Not Found
[2023-03-17T06:23:01.200Z]   IDEs:
[2023-03-17T06:23:01.200Z]     Android Studio: Not Found
[2023-03-17T06:23:01.200Z]     Xcode: 14.2/14C18 - /nix/store/i4qw4s2n4650cwkzmfxq1zm3yav26x1r-xcode-wrapper-13.3-plus/bin/xcodebuild
[2023-03-17T06:23:01.200Z]   Languages:
[2023-03-17T06:23:01.200Z]     Java: Not Found
[2023-03-17T06:23:01.200Z]   npmPackages:
[2023-03-17T06:23:01.200Z]     @react-native-community/cli: Not Found
[2023-03-17T06:23:01.200Z]     react: 18.0.0 => 18.0.0 
[2023-03-17T06:23:01.200Z]     react-native: 0.69.8 => 0.69.8 
[2023-03-17T06:23:01.200Z]     react-native-macos: Not Found
[2023-03-17T06:23:01.200Z]   npmGlobalPackages:
[2023-03-17T06:23:01.200Z]     *react-native*: Not Found
```

siddarthkay commented:
```log
/Users/siddarthkumar/code/experiments/status-mobile/modules/react-native-status/ios/RCTStatus/RCTStatus.h:4:9: fatal error: 'Statusgo.h' file not found
#import "Statusgo.h"
        ^~~~~~~~~~~~
1 error generated.
```

the latest error I get locally is related to : https://github.com/status-im/status-mobile/issues/2246

siddarthkay commented:
Missing Statusgo error was fixed by destroying and installing pods on local environment, Now the sole focus is the error I see both on local and on CI 

```
In file included from /Users/siddarthkumar/code/experiments/status-mobile/modules/react-native-status/ios/RCTStatus/RCTStatus.m:2:
/Users/siddarthkumar/code/experiments/status-mobile/node_modules/react-native/ReactCommon/react/config/ReactNativeConfig.h:10:10: fatal error: 'string' file not found
#include <string>
         ^~~~~~~~
1 error generated.
```

I see we use a fork of `react-native-config` library so it should be easier to play with and fix :)

siddarthkay commented:
> we should go straight to 0.71 and new architecture

So 0.71.4 does work for IOS but without the new architecture.

<img width="834" alt="Screenshot_2023-03-19_at_10 02 42_AM" src="https://user-images.githubusercontent.com/64726664/226154569-2eb3ce29-7355-4be6-b7c1-c79855bb1e55.png">


We will have to modify the native modules project and that seems like a sizeable refactor to me at this point, will open up a new issue over there and outline the challenges with using Fabric UI engine and replacing the bridge of React Native.

Next steps are to get the Android side of changes merged in to work with 0.71.4 :)

siddarthkay committed:
(0.71.4) RN Android Upgrade

siddarthkay committed:
(0.71.4) Android - fixes

`ndkVersion`-> "23.1.7779620"

android: fix missing gradle dependencies

nix: moar android fixes

add dependency for AsyncStorage

siddarthkay committed:
(0.71.4) IOS Fixes

refactor to use latest react-native-config library

add tsconfig & remove flowconfig

upgrade `react-native-redash` to `18.1.0`

flexsurfer commented:
hey @siddarthkay could you pls give a short update on latest status of this PR, i see you created new PR with older version

siddarthkay commented:
> no luck with 0.71.4

no luck with any version after 0.67.5!
At 0.71.4 Only ray of hope was that IOS builds would be successful on CI only 
We could not even get those builds to run on simulator.

The transition to enable new architecture and use newer navigation library and newer reanimated library was too much to handle all together.
Hot reloading was also not working so i had to rebuild application each time and it was just a mess to fix issues on.

So I figured its best to merge what is stable (i.e 0.67) into develop and observe if it breaks any existing functionality and isolate each issue and work on it separately. 
I won't be giving up on the upgrade that easily I want to tackle each issue gradually.

OmarBasem commented:
Hi @siddarthkay,
Keep in mind that upgrading to v0.70+ and enabling new architecture are 2 different issues. Lots of third-party libraries do not yet support new-arch as it is still in a transition phase, and will probably remain so as long as it is optional from react-native. I think we should focus on upgrading to v0.70+, and then see the possibility of enabling the `new-arch` after that. We will still reap lots of benefits by upgrading to v0.70+ with or without `new-arch`.

flexsurfer commented:
hey @siddarthkay should we close this or rename maybe, because we are not on 0.63.3 anymore ?

flexsurfer commented:
feel free to reopen when work is ongoing
